provide and use compiled in test certs (only) in tests

tbg opened the pull request:
this allows tests to run without external dependencies while avoiding compiled-in testdata in the main build.

MBerhault commented on security/certs_test.go:
```diff
@@ -28,7 +28,8 @@ import (
 // This is just the mechanics of certs generation.
-func TestGenerateCerts(t *testing.T) {
+// TODO(tschottdorf) make this test use the actual file system.
+func disabledTestGenerateCerts(t *testing.T) {
```
shouldn't this work is you call ResetReadFileFn?

tschottdorf replied:
yes, hence the todo. It's done now.

MBerhault commented on server/cli/cli_test.go:
```diff
@@ -49,8 +48,8 @@ func (c cliTest) Run(line string) {
 	args = append(args, fmt.Sprintf("-addr=%s", c.ServingAddr()))
-	// Always load server certs. Not sure if this path is a good assumption though.
-	args = append(args, fmt.Sprintf("-certs=%s", security.EmbeddedPrefix+"test_certs"))
+	// Always load test certs. Not sure if this path is a good assumption though.
```
you can remove the part about the path, that was when I used ${GOPATH}/src/...

tschottdorf replied:
done.

tschottdorf commented:
PTAL @MBerhault

MBerhault commented:
LGTM. assuming no test running were missed.

tbg committed:
provide and use compiled in test certs (only) in tests

this allows tests to run without external dependencies,
while avoiding compiled-in testdata in the main build.

spencerkimball commented on rpc/testing.go:
```diff
@@ -26,7 +26,7 @@ import (
 // NewTestContext returns a rpc.Context for testing.
 func NewTestContext(t *testing.T) *Context {
-	tlsConfig, err := security.LoadTLSConfigFromDir(security.EmbeddedPrefix + "test_certs")
+	tlsConfig, err := security.LoadTLSConfigFromDir("test_certs")
```
Instead of leaking this constant "test_certs" directory, why not just have another security method called: `security.LoadTLSConfigForTests`? Then in security:

    func LoadTLSConfigForTests() (*TLSConfig, error) {
      return LoadTLSConfigFromDir("test_certs")
    }

spencerkimball replied:
Or make it an exported constant in the `security` package.

MBerhault replied:
I'm currently refactoring some of this with much fewer TLS setups. I'll move the test path to an exported const.

tschottdorf replied:
@MBerhault is refactoring the stuff anyways, I'll wait for the dust to settle. What's more worrying is that we currently **do** compile in the test certs everywhere, because it's used in the `rpc/testing` package. Will change that though.
